pycaret/internal/pycaret_experiment/time_series_experiment.py

```diff
@@ -484,56 +486,86 @@ def _check_and_set_fh(
+        (2) the derived seasonal periods for which seasonality is detected.
 
         Parameters
         ----------
         data : pd.DataFrame
             Data used index can be used to extract the seasonal period information
-        seasonal_period : Optional[Union[int, str]]
+        seasonal_period : Optional[Union[List[Union[int, str]], int, str]]
             Seasonal Period specified by user
 
         Raises
         ------
         ValueError
-            If seasonal period is provided but is not of type int or string
+            If seasonal period is provided but its values are not of type int or string
         """
         self.logger.info("Set up Seasonal Period.")
 
         # sktime is an optional dependency
         from sktime.utils.seasonality import autocorrelation_seasonality_test
 
         if seasonal_period is None:
-            index_freq = data.index.freqstr
-            self.seasonal_period = get_sp_from_str(str_freq=index_freq)
-        else:
-            if not isinstance(seasonal_period, (int, str)):
-                raise ValueError(
-                    f"seasonal_period parameter must be an int or str, got {type(seasonal_period)}"
-                )
+            seasonal_period = data.index.freqstr
 
-            if isinstance(seasonal_period, str):
-                self.seasonal_period = get_sp_from_str(str_freq=seasonal_period)
-            else:
-                self.seasonal_period = seasonal_period
+        if not isinstance(seasonal_period, list):
+            seasonal_period = [seasonal_period]
+        seasonal_period = [self._convert_sp_to_int(sp) for sp in seasonal_period]
 
         # check valid seasonal parameter
-        self.seasonality_present = autocorrelation_seasonality_test(
-            data[self.target_param], self.seasonal_period
-        )
+        seasonality_test_results = [
+            autocorrelation_seasonality_test(data[self.target_param], sp)
+            for sp in seasonal_period
+        ]
+        self.seasonality_present = any(seasonality_test_results)
+        sp_values_and_test_result = zip(seasonal_period, seasonality_test_results)
 
         # What seasonal period should be used for modeling?
-        self.sp_to_use = self.seasonal_period if self.seasonality_present else 1
+        self.all_sp_values = [
+            sp
+            for sp, seasonality_present in sp_values_and_test_result
+            if seasonality_present
+        ] or [1]
+        self.sp_to_use = self.all_sp_values[0]
+        self.seasonal_period = seasonal_period[0]
+
+    def _convert_sp_to_int(self, seasonal_period):
+        """Derives the seasonal period specified by either:
+            (1) Extracting it from the seasonal_period if it is of type string, or
+            (2) Using seasonal_period as is if it is of type int.
+
+        Parameters
+        ----------
+        seasonal_period : Optional[Union[List[Union[int, str]], int, str]]
```

Comment:
In this method, it will only be of type int or str (and not List) - correct?

Reply:
Oops, sorry! You are right. I got confused when reviewing the type hints.